### DISCUSSION TITLE
Control switch of Shelly plus 1 over Bluetooth

### DISCUSSION THREAD

**Discussion by rept** [arendst/Tasmota#13288] [Ideas]
Shelly released the Shelly plus 1 based on the ESP32 which supports bluetooth.  Currently they only offer the bluetooth for discovery.  Are there any plans to enable the bluetooth also for controlling the switch: turning the switch on and off.  What would the development effort be for such a feature?

I've been looking for an alternative to Shelly based on bluetooth (instead of Wifi) with the same price point.

**Comment by sfromis** [comment]
There is no "master plan" for things to be implemented. Lots of features arise from someone with the inclination, skill and will sitting down to do the coding, and submitting a PR. My guess is that adding a BLE client feature would not be a small effort, but it could depend on availability of some existing library suitable for having a Tasmota driver on top. There are also questions about how well it could work with existing Bluetooth-oriented apps. I find it hard to see Bluetooth as a goal in itself, unless able to fit into existing frameworks. Of course, if someone wants to develop mobile apps for the purpose, but .....

**Comment by barbudor** [comment]
@rept what would be the goal of Bluetooth control?

Bluetooth, unlike wifi, needs pairing between each device and each phone that needs to talk to it. When you would like to control your swich you would have to connect your phone to the switch. And that connection would just last while your phone is around. That doesn't sound more practical than having wifi AP on each device instead of every device connected to the same AP.

Some other groups have added Bluetooth support in their firmware only for the configuration phase. This has not been a chosen path for Tasmota so far.

Now if you are looking to use Bluetooth as a network like mesh. May be because you don't feel Wifi should be used for that for example. Then there are IMHO much better technology around for that like Zigbee which still can benefit of a Tasmota gateway.

Just my 2c. Happy to get your feedback.

**Reply by rept** [reply]
Hi @barbudor indeed my request would seem odd if you would consider it from a phone...  

In our case we have a smart lamp that is able to detect elderly people that fall.  It has Bluetooth on board and we use that to link with other devices (for example smart scale, smart lock, etc...).  The lamp also has a SIM card and acts as a hub.  Our smart lamp replaces the existing lamp.  That also means it is powered at all times.  Sometimes when we install it in existing homes there is a second (normal) lamp on the same circuit.  Since the circuit is powered all the time, we use Shelly to control that second lamp to switch on and off together with our lamp.  However Shelly doesn't work on 5Ghz, sometimes Wifi is unstable, etc so this isn't ideal.

I hope this explains our use case a bit.

**Reply by barbudor** [reply]
Thanks, I understand the goal and I just checked your web site. Very interresting idea and I could consider it personnaly.

Tasmota is really centered around Wifi and MQTT.
Bluetooth is used to received sensors, or in some case talk to Thermostats.
Control from bluetooth hasn't been considered as far as I know but that's an open source project, so any one can contribute and if you want to implement then why not merging it.

**Reply by Twisted7ech** [reply]
If the bluetooth on the plus 1 is design to receive data from sensors, would it be possible to use signal strength of known devices to trigger the switch?

**Reply by sfromis** [reply]
As the usual payloads for BLE sensors includes RSSI, nothing stops you from creating an automation based on this. However, BLE sensors usually are not transmitting constantly, but at intervals to save battery, meaning that there's a high risk of rather slow reaction time. Also it is not very safe to use the signal strength as any kind of reliable indicator of distance.

**Reply by petterjohansson** [reply]
If you want to control a garage door outside wifi range it would be nice if you could use Bluetooth.

**Reply by barbudor** [reply]
I see many drawbacks for that:
- The Shelly 1 Plus being single-core is not the best target for control + wifi  + bluetooth
  Workaround: Sonoff Dual R3 has a dual-core ESP32 and might be more suitable
- You need to find a source for bluetooth messages that will work.
  Question: Is there any mobile app that allows you to create dash board with buttons to send custom bluetooth messages ?
- Bluetooth messages without no kind of authentication process are likely to be sniffed and re-used
  Possible workaround : Have the app generate a rolling code / OTP that then need to be decoded/checked on Tasmota or back-end

As far as a mobile is involved, it is much more easier to build a secure infrastructure with a VPN or using cloud based (free) MQTT server as a relay for limited operations (not for your main MQTT broker)

**Reply by sfromis** [reply]
For the "outer range" issues, I'd generally prefer to extend the wifi range. In some cases, you can set up an existing Tasmota device with the range extender feature providing an extra low-speed hotspot.

**Comment by xaris33** [comment]
Could the Bluetooth possibly get used for room presence detection? It would be ideal if every room had a switch.

**Reply by barbudor** [reply]
More or less but as bluetooth can cross walls, devices can be detected by multiple ESP32
So you would need some processing on a backend, aggregating information such as reception level, to triangulate or filter the information

**Reply by schildbach** [reply]
I think this could be a useful feature. Many of us run a backend already, e.g. with MQTT or Node-Red.

Even just the mere presence of a device could be useful, e.g. for detecting guests on premise.

**Reply by christf** [reply]
I am looking for a way to do presence detection as well and it would seem like a waste to litter bluetooth devices across the place in addition to the bluetooth devices that are already built into the shellies.

**Reply by barbudor** [reply]
You can install tasmota32**-bluetooth on any esp32** (except solo1)
